go.mod

```diff
@@ -1,8 +1,8 @@
 module github.com/ethereum/go-ethereum
 
-go 1.22
+go 1.22.0
 
-toolchain go1.22.0
+toolchain go1.23.5
```

Comment:
Any particular reason for bumping the toolchain version here?

What's the policy for bumping the go and toolchain version for our codebase?

Probably we should also bump the go to `1.23.5`?